src/algorithms/rhythm/tempocnn.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#include "tempocnn.h"
+
+using namespace std;
+using namespace essentia;
+using namespace standard;
+
+const char* TempoCNN::name = "TempoCNN";
+const char* TempoCNN::category = "Rhythm";
+const char* TempoCNN::description = DOC(
+  "This algorithm makes predictions using Hendrik Schreiber's TempoCNN-based models.\n"
+  "Internally, this algorithm is a wrapper to aggregate the predictions generated by TensorflowPredictTempoCNN. "
+  "localTempo is a vector containing the most likely BPM estimated each ~6 seconds by default. localTempoProbabilities contains the probabilities attached to the tempo estimations and can be used as a confidence measure.\n"
+  "globalTempo is an aggregation of localTempo using an `aggregationMethod`. We strongly recommend to use majority voting when assuming constant tempo in the input audio.\n"
+  "See TensorflowPredictTempoCNN for details about the rest of parameters.\n"
+  "The recommended pipeline is as follows:\n"
+  "  MonoLoader(sampleRate=11025) >> TempoCNN"
+  "\n"
+  "Note: This algorithm does not make any check on the input model so it is the user's responsibility to make sure it is a valid one."
+  "\n"
+  "References:\n"
+  "  [1] Hendrik Schreiber, Meinard Müller, A Single-Step Approach to Musical Tempo Estimation Using a Convolutional Neural Network Proceedings of the 19th International Society for Music Information Retrieval Conference (ISMIR), Paris, France, Sept. 2018.\n"
+  "  [2] Hendrik Schreiber, Meinard Müller, Musical Tempo and Key Estimation using Convolutional Neural Networks with Directional Filters Proceedings of the Sound and Music Computing Conference (SMC), Málaga, Spain, 2019.\n"
+  "  [3] Original models and code at https://github.com/hendriks73/tempo-cnn\n"
+  "  [4] Supported models at https://essentia.upf.edu/models/");
+
+
+// Hendrik's models output a softmax distribution of length 256.
```

Comment:
May be we should refer to TempoCNN in the comments or just "the models", especially if he creates more models in the future.